dali/operators/image/convolution/gaussian_blur.cc

```diff
@@ -0,0 +1,309 @@
+    auto& output = ws.template OutputRef<CPUBackend>(0);
+    auto in_shape = input.shape();
+    auto& thread_pool = ws.GetThreadPool();
+
+    // TODO(klecki): Swap to priority thread_pool task scheduling
+    std::vector<std::pair<int64_t, int>> volume_idx_vec(input.shape().num_samples());
+    for (int i = 0; i < input.shape().num_samples(); i++) {
+      const auto& shape = input[i].shape();
+      auto elem_volume = volume(shape.begin() + dim_desc_.usable_axes_start, shape.end());
+      volume_idx_vec[i] = {elem_volume, i};
+    }
+    std::sort(volume_idx_vec.begin(), volume_idx_vec.end(),
+              std::greater<std::pair<int64_t, int>>());
+
+    for (const auto& sample_order : volume_idx_vec) {
+      auto sample_idx = sample_order.second;
+      int seq_elements = 1;
+      int64_t stride = 0;
+      if (dim_desc_.is_sequence) {
+        const auto& shape = input[sample_idx].shape();
+        seq_elements = volume(shape.begin(), shape.begin() + dim_desc_.usable_axes_start);
+        stride = volume(shape.begin() + dim_desc_.usable_axes_start, shape.end());
+      }
+      for (int elem_idx = 0; elem_idx < seq_elements; elem_idx++) {
+        thread_pool.DoWorkWithID(
+            [this, &input, &output, sample_idx, elem_idx, stride](int thread_id) {
+              auto gaussian_windows = windows_[sample_idx].GetWindows();
+              auto elem_shape = input[sample_idx].shape().template last<ndim>();
+              auto in_view = TensorView<StorageCPU, const In, ndim>{
+                  input[sample_idx].template data<In>() + stride * elem_idx, elem_shape};
+              auto out_view = TensorView<StorageCPU, Out, ndim>{
+                  output[sample_idx].template mutable_data<Out>() + stride * elem_idx, elem_shape};
+              // I need a context for that particular run (or rather matching the thread &
+              // scratchpad)
+              auto ctx = ctx_;
+              kmgr_.Run<Kernel>(thread_id, sample_idx, ctx, out_view, in_view, gaussian_windows);
+            });
+      }
+    }
+    thread_pool.WaitForWork();
+  }
+
+ private:
+  OpSpec spec_;
+  int batch_size_ = 0;
+  DimDesc dim_desc_;
+
+  kernels::KernelManager kmgr_;
+  kernels::KernelContext ctx_;
+
+  std::vector<GaussianBlurParams<axes>> params_;
+  std::vector<GaussianWindows<axes>> windows_;
+};
+
+template <>
+bool GaussianBlur<CPUBackend>::SetupImpl(std::vector<OutputDesc>& output_desc,
+                                         const workspace_t<CPUBackend>& ws) {
+  const auto& input = ws.template InputRef<CPUBackend>(0);
+  auto layout = input.GetLayout();
+  auto dim_desc = ParseAndValidateDim(input.shape().sample_dim(), layout);
+  dtype_ = dtype_ != DALI_NO_TYPE ? dtype_ : input.type().id();
+  DALI_ENFORCE(dtype_ == input.type().id() || dtype_ == DALI_FLOAT,
+               "Unsupported output data type. Output type can be same as input (inferred "
+               "automatically) or set to float.");
```

Comment:
A bit convoluted sentence. Alternative wording:
```suggestion
               "Output data type must be same as input, FLOAT or skipped (defaults to input type)");
```

Reply:
Done.